src/components/MultipleFileInputField/MultipleFileInputField.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Form, FormControl, IconButton,
+} from '@edx/paragon';
+import classNames from 'classnames';
+import * as FontAwesome from '@fortawesome/free-solid-svg-icons/faTimes';
+import { getSizeInBytes, formatBytes } from './utils';
+
+const MultipleFileInputField = ({
+  input,
+  label,
+  type,
+  description,
+  disabled,
+  required,
+  className,
+  id,
+  meta: { touched, error },
+  ...props
+}) => {
+  const hasError = !!(touched && error);
+  const [size, setSize] = useState('0');
+  const { name: inputName, onChange: inputOnChange, value: inputValues } = input;
+
+  const icon = FontAwesome.faTimes;
+  const handleInputChange = (e) => {
+    const file = e.target.files[0];
+    const sizeOfPreviousFiles = getSizeInBytes(size);
+    const totalSize = sizeOfPreviousFiles + file.size;
+    if (totalSize > 2097152) {
+      // TODO: displau some error if size gets large
+    } else {
+      const fileReader = new FileReader();
+      fileReader.onload = () => {
+        const { name: fileName, size: fileSize } = file;
+        const { result: fileContents } = fileReader;
+        inputOnChange([...inputValues, { name: fileName, size: fileSize, contents: fileContents }]);
+      };
+      fileReader.readAsText(file);
+    }
+  };
+
+  useEffect(() => {
+    let sizeOfFiles = 0;
+    if (inputValues && inputValues.length > 0) {
+      sizeOfFiles = inputValues.reduce((total, file) => total + file.size, 0);
+    }
+    setSize(formatBytes(sizeOfFiles));
+  }, [inputValues]);
+
+  const handleFileRemoveClick = (index) => {
+    inputOnChange(inputValues.filter((e, i) => i !== index));
+  };
+
+  return (
+    <Form.Group id={id} className={className}>
+      <Form.Label>
+        {label}
+      </Form.Label>
+      <span>{size !== '0' ? ` - ${size}` : null}</span>
+      <br />
+      <Form.Label
+        className={classNames(
+          'choose-file-btn',
+          'btn',
+          'btn-link',
+          'p-1',
+          'm-0',
+          'ml-2',
+        )}
+        htmlFor="1234"
+      >Select files
+      </Form.Label>
+      <Form.Control
+        name={inputName}
+        id="1234"
+        type="file"
+        value=""
+        {...props}
+        onChange={handleInputChange}
+        style={{ position: 'absolute', width: '0px', height: '0px' }}
+      />
+
+      {hasError && <FormControl.Feedback type="invalid">{error}</FormControl.Feedback>}
+      {description && <Form.Text>{description}</Form.Text>}
+      {
+        inputValues && inputValues.map((e, i) => (
+          <div className="border rounded p-1 m-1">
+            <small>{`${e.name } - ${ formatBytes(e.size)}`}</small>
+            <IconButton className="mr-2" icon={icon} onClick={() => handleFileRemoveClick(i)} variant="danger" />
+          </div>
+        ))
+      }
+    </Form.Group>
+  );
+};
+
+MultipleFileInputField.defaultProps = {
+  description: null,
```

Comment:
I think it is better to make these attributes required if they do not have a usable default value.

Reply:
done